File: Patrones Comportamiento/Observer/Observer.py
# Observer
class Subject:
    def __init__(self):
        self.observers = []

    def attach(self, o):
        self.observers.append(o)

    def notify(self):
        for o in self.observers:
            o.update()

class Observer:
    def update(self):
        print("Observador notificado")

s = Subject()
s.attach(Observer())
s.notify()

File: Patrones Comportamiento/Observer/Observer.js
// Observer
class Subject {
  constructor() {
    this.observers = [];
  }
  attach(o) {
    this.observers.push(o);
  }
  notify() {
    this.observers.forEach(o => o.update());
  }
}

class Observer {
  update() {
    console.log("Observador notificado");
  }
}

const s = new Subject();
s.attach(new Observer());
s.notify();
